scripts/utils/git/index.js

```diff
@@ -0,0 +1,47 @@
+import diff from 'deep-diff';
+import simpleGit from 'simple-git';
+
+const uniq = array => Array.from(new Set(array));
+
+const filePathToServiceId = filePath => filePath
+  .replace('declarations/', '')
+  .replace('.history', '')
+  .replace('.filters', '')
+  .replace('.json', '');
+
+const getModifiedData = async instancePath => {
+  const git = simpleGit(instancePath, { maxConcurrentProcesses: 1 });
+
+  const modifiedFilePaths = await (await git.diff([ '--name-only', 'HEAD', 'main', '--', './declarations' ])).split('\n');
+
+  return { git, modifiedFilePaths, modifiedServiceIds: uniq(modifiedFilePaths.filter(Boolean).map(filePathToServiceId)) };
+};
+
+export const getModifiedServiceDocumentTypes = async instancePath => {
+  const { git, modifiedFilePaths, modifiedServiceIds } = await getModifiedData(instancePath);
+
+  const servicesDocumentTypes = modifiedServiceIds.reduce((acc, val) => ({ ...acc, [val]: [] }), {});
+
+  await Promise.all(modifiedFilePaths.map(async modifiedFilePath => {
+    try {
+      const serviceId = filePathToServiceId(modifiedFilePath);
+      const fileInMain = JSON.parse(await git.show([`main:${modifiedFilePath}`]));
+      const fileModified = JSON.parse(await git.show([`HEAD:${modifiedFilePath}`]));
+      const modifiedDocumentTypes = diff.diff(fileInMain, fileModified).map(({ path }) => (path[0] === 'documents' ? path[1] : path[0]));
+
+      servicesDocumentTypes[serviceId] = uniq(modifiedDocumentTypes);
```

Comment:
Does not work when service file is modified outside the `documents` key. (For example when changing service name)

```suggestion
const modifiedDocumentTypes = DeepDiff.diff(fileInMain, fileModified).reduce((acc, { path }) => {
        if (modifiedFilePath.includes('.history')) {
          acc.add(path[0]);
        } else if (path[0] == 'documents') {
          acc.add(path[1]);
        }

        return acc;
      }, new Set());

      servicesDocumentTypes[serviceId] = Array.from(modifiedDocumentTypes);
```